Subject: Add EcoNet bad block table

The EcoNet BBT/BMT is used for resolving bad blocks in the flash.
It is implemented in the EcoNet bootloader so you cannot safely
access flash without using it.

Signed-off-by: Caleb James DeLisle <cjd@cjdns.fr>
---
--- a/drivers/mtd/nand/Makefile
+++ b/drivers/mtd/nand/Makefile
@@ -3,7 +3,7 @@
 nandcore-objs := core.o bbt.o
 obj-$(CONFIG_MTD_NAND_CORE) += nandcore.o
 obj-$(CONFIG_MTD_NAND_ECC_MEDIATEK) += ecc-mtk.o
-obj-$(CONFIG_MTD_NAND_MTK_BMT)	+= mtk_bmt.o mtk_bmt_v2.o mtk_bmt_bbt.o mtk_bmt_nmbm.o
+obj-$(CONFIG_MTD_NAND_MTK_BMT)	+= mtk_bmt.o mtk_bmt_v2.o mtk_bmt_bbt.o mtk_bmt_nmbm.o en75_bmt.o
 obj-$(CONFIG_SPI_QPIC_SNAND) += qpic_common.o
 obj-$(CONFIG_MTD_NAND_QCOM) += qpic_common.o
 obj-y	+= onenand/
--- a/drivers/mtd/nand/mtk_bmt.h
+++ b/drivers/mtd/nand/mtk_bmt.h
@@ -77,6 +77,7 @@ extern struct bmt_desc bmtd;
 extern const struct mtk_bmt_ops mtk_bmt_v2_ops;
 extern const struct mtk_bmt_ops mtk_bmt_bbt_ops;
 extern const struct mtk_bmt_ops mtk_bmt_nmbm_ops;
+extern const struct mtk_bmt_ops en75_bmt_ops;
 
 static inline u32 blk_pg(u16 block)
 {
--- a/drivers/mtd/nand/mtk_bmt.c
+++ b/drivers/mtd/nand/mtk_bmt.c
@@ -421,6 +421,8 @@ int mtk_bmt_attach(struct mtd_info *mtd)
 		bmtd.ops = &mtk_bmt_nmbm_ops;
 	else if (of_property_read_bool(np, "mediatek,bbt"))
 		bmtd.ops = &mtk_bmt_bbt_ops;
+	else if (of_property_read_bool(np, "econet,bmt"))
+		bmtd.ops = &en75_bmt_ops;
 	else
 		return 0;
 
